src/cov/cov-data.c

```diff
@@ -0,0 +1,823 @@
+//
+//  Copyright (C) 2013-2023  Nick Gasson
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+#include "util.h"
+#include "array.h"
+#include "cov/cov-api.h"
+#include "cov/cov-data.h"
+#include "ident.h"
+#include "lib.h"
+#include "option.h"
+#include "tree.h"
+#include "type.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <libgen.h>
+
+//#define COVER_DEBUG_EMIT
+//#define COVER_DEBUG_DUMP
+//#define COVER_DEBUG_SCOPE
+//#define COVER_DEBUG_MERGE
+//#define COVER_DEBUG_CALLBACK
```

Comment:
I think `COVER_DEBUG_EXCLUDE` should go to `cover_exclude.c` and `COVER_DEBUG_CALLBACK` to `rt/cover.c`